src/MKRWAN.h

```diff
@@ -671,22 +708,21 @@ class LoRaModem : public Stream
     if (!autoBaud()) {
       return false;
     }
-    sendAT(GF("+REBOOT"));
-    if (waitResponse(10000L, "+EVENT=0,0") != 1) {
+    sendAT(GF(AT_RESET));
+    if (waitResponse(10000L, GF(AT_EVENT) GF(AT_EQ) "0,0") != 1) {
```

Comment:
Hello @flhofer, I'm having a failure when calling `modem.restart()`: `waitResponse()` is timing out, probably because of a not handled case.
Could you please double check it on your side? Thanks!

Reply:
Hi @giulcioffi 
sorry for the late reply. Found and fixed the reset error. My tests showed an unhandled situation in the library. Here is the log from my device with debug output.
```
Reset
### AT: 
### Data string: +OK
### AT: +REBOOT
### Data string: +EVENT          <-- valid terminator '=', but string does not match '+EVENT=0,0'
### Unhandled: +EVENT=0,0   <-- parsing timeout. we didn't receive any of the valid terminators (=,+,\r)
### AT: 
### Data string: +EVENT
### Data string: +EVENT=0,0
### Data string: +EVENT=0,0+OK.   <-- AT triggered new parsing.. now reacting to +OK
### AT: +DEV ?
### Data string: +DEV
### AT: +VER ?
```
Indeed, the library's waitResponse only parses `["=", "+", "\r"]` but the event message did not terminate with either of them. Thus, the check in the library timed out. I now added the `\r` to the reboot event message. Tested and works fine.
😃 ready to merge?